### PR TITLE
Add oauth backup and restore step

### PR DESCRIPTION
**Pull request by stuartchuan**
Add the OAuth backup and restore step.

### CODE REVIEW

**Comment by stuartchuan** [comment]
@pruan-rht @yapei please help review, thanks.

**Comment by akostadinov** [comment]
how about using the step `hidden recreate cluster resource( after scenario)?` ?

**Comment by stuartchuan** [comment]
The OAuth CRD is different with others, it could not be deleted, and also the generation field must be  accumulated.

File: features/step_definitions/oauth.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+Given /^the #{QUOTED} oauth CRD is restored after scenario$/ do |name|
+  ensure_admin_tagged
+  org_oauth = {}
+  @result = admin.cli_exec(:get, resource: 'oauth', resource_name: name, o: 'yaml')
+  if @result[:success]
+     org_oauth['spec'] = @result[:parsed]['spec']
+     require 'pry'; binding.pry
```
[the remark this answers is not in the document]
Comment (pruan-rht) [reply features/step_definitions/oauth.rb]: also indentation issue.  I'll file a PR to fix them.

[the remark this answers is not in the document]
Comment (stuartchuan) [reply features/step_definitions/oauth.rb]: sorry, my mistake.

[the remark this answers is not in the document]
Comment (pruan-rht) [reply features/step_definitions/oauth.rb]: @stuartchuan, no worries, thanks for filing the PR, the fix has been merged already, so it's all good...

**Pull request merged**